From 06da92170f7fbb66a1557e3317e186dfcca688ea Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 12 Aug 2014 19:24:26 +0000
Subject: [PATCH 0489/1000] MANTIS 0033302 change mstarter: use only defined
 system environment variables for substitution

massai_text.lib
V1.0.0.6

- use these system environment variables
   -ALLUSERSPROFILE
   -APPDATA
   -COMPUTERNAME
   -ComSpec
   -CommonProgramFiles
   -CommonProgramW6432
   -HOMEDRIVE
   -HOMEPATH
   -HOMESHARE
   -LOCALAPPDATA
   -LOGONSERVER
   -PROGRAMFILES
   -PSModulePath
   -PUBLIC
   -Path
   -ProgramData
   -ProgramFiles
   -ProgramW6432
   -SYSTEMDRIVE
   -SYSTEMROOT
   -TEMP
   -TMP
   -USERDNSDOMAIN
   -USERDOMAIN
   -USERNAME
   -USERPROFILE
   -windir



git-svn-id: svn://localhost/SelfServiceCommon/trunk@709 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Interfaces/inc/text/readEnv.h | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/SelfServiceCommon/Interfaces/inc/text/readEnv.h b/SelfServiceCommon/Interfaces/inc/text/readEnv.h
index 23d3ab23..2ba67037 100644
--- a/SelfServiceCommon/Interfaces/inc/text/readEnv.h
+++ b/SelfServiceCommon/Interfaces/inc/text/readEnv.h
@@ -13,15 +13,9 @@ namespace mText
 {
   typedef std::map<std::string, std::string> Environment;
 
-  class ReadEnvironment
-  {
-    public:
-      virtual ~ReadEnvironment();
-      virtual Environment getEnv() = 0;
-  };
-
-  ReadEnvironment* getConfigReader(const std::string& authentification, MASSAI::trace::TraceInterface& trc);
-  ReadEnvironment* getEnvironmentVariablesReader(MASSAI::trace::TraceInterface& trc);
+  Environment getConfigVariables(const std::string& authentification, MASSAI::trace::TraceInterface& trc);
+  Environment getFullEnvironment(MASSAI::trace::TraceInterface& trc);
+  Environment getSystemEnvironment(MASSAI::trace::TraceInterface& trc);
 }
 
 #endif
-- 
2.41.0.windows.1

